systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/PodUtils.java

```diff
@@ -144,19 +144,19 @@ public static void waitForPod(String name) {
     public static void waitForPodDeletion(String name) {
         LOGGER.info("Waiting when Pod {} will be deleted", name);
 
-        if (kubeClient().listPodsByPrefixInName(name).size() != 0) {
-            TestUtils.waitFor("Pod " + name + " could not be deleted", Constants.POLL_INTERVAL_FOR_RESOURCE_DELETION, Constants.TIMEOUT_FOR_POD_DELETION,
-                () -> {
-                    Pod pod = kubeClient().getPod(name);
-                    if (pod == null) {
-                        return true;
-                    } else {
+        TestUtils.waitFor("Pod " + name + " could not be deleted", Constants.POLL_INTERVAL_FOR_RESOURCE_DELETION, Constants.TIMEOUT_FOR_POD_DELETION,
```

Comment:
I think this is a bit confusing. The method name says `waitForPodDeletion` - so I expect it that not delete anything, just wait for it to be deleted. Yet the message `Pod xyz could not be deleted` suggests this is actually a method deleting the pod? I think we should rename this method to something more matching for what it does ... e.g. `deletePodWithWait`, `deletePodSync` etc.

If you wanna do that in separate PR, that should be fine. But it should be fixed because when you read in the code `waitForPodDeletion` you do not expect it to delete anything. I also don't think this makes any sense form the test perspective. You want the CO to delete the pod. Not delete it your self.

Reply:
The reason for that was cause from time to time we faced pods, which were stuck in terminating state. Maybe we can add here condition to force delete pods only if it's stuck.

Reply:
I can do that in separate PR, thanks for the comment